kafka/channel/pkg/apis/messaging/v1alpha1/kafka_channel_conversion.go

```diff
@@ -0,0 +1,179 @@
+		statusAddressable := &duckv1.Addressable{
+			URL: nil,
+		}
+		if source.Status.AddressStatus.Address != nil {
+			if err := source.Status.AddressStatus.Address.ConvertTo(ctx, statusAddressable); err != nil {
+				return err
+			}
+		}
+
+		subscribableStatus := eventingduckv1.SubscribableStatus{
+			Subscribers: nil,
+		}
+		if source.Status.SubscribableStatus != nil && len(source.Status.SubscribableStatus.Subscribers) > 0 {
+			subscribableStatus.Subscribers = make([]eventingduckv1.SubscriberStatus, len(source.Status.SubscribableStatus.Subscribers))
+			for i, ss := range source.Status.SubscribableStatus.Subscribers {
+				subscribableStatus.Subscribers[i] = eventingduckv1.SubscriberStatus{
+					UID:                ss.UID,
+					ObservedGeneration: ss.ObservedGeneration,
+					Ready:              ss.Ready,
+					Message:            ss.Message,
+				}
+			}
+		}
+
+		subscribableSpec := eventingduckv1.SubscribableSpec{
+			Subscribers: nil,
+		}
+		if source.Spec.Subscribable != nil && len(source.Spec.Subscribable.Subscribers) > 0 {
+			subscribableSpec.Subscribers = make([]eventingduckv1.SubscriberSpec, len(source.Spec.Subscribable.Subscribers))
+			for i, ss := range source.Spec.Subscribable.Subscribers {
+				delivery := &eventingduckv1.DeliverySpec{}
+				if err := ss.Delivery.ConvertTo(ctx, delivery); err != nil {
+					return err
+				}
+
+				subscribableSpec.Subscribers[i] = eventingduckv1.SubscriberSpec{
+					UID:           ss.UID,
+					Generation:    ss.Generation,
+					SubscriberURI: ss.SubscriberURI.DeepCopy(),
+					ReplyURI:      ss.ReplyURI,
+					Delivery:      delivery,
+				}
+			}
+		}
+
+		sink.ObjectMeta = source.ObjectMeta
+		sink.Spec = v1beta1.KafkaChannelSpec{
+			NumPartitions:     source.Spec.NumPartitions,
+			ReplicationFactor: source.Spec.ReplicationFactor,
+			ChannelableSpec: eventingduckv1.ChannelableSpec{
+				SubscribableSpec: subscribableSpec,
+				// no delivery in v1alpha1
+				Delivery: nil,
+			},
+		}
+		sink.Status = v1beta1.KafkaChannelStatus{
+			ChannelableStatus: eventingduckv1.ChannelableStatus{
+				Status: source.Status.Status,
+				AddressStatus: duckv1.AddressStatus{
+					Address: statusAddressable,
+				},
+				SubscribableStatus: subscribableStatus,
+				// no DeadLetterChannel in v1alpha1
+				DeadLetterChannel: nil,
+			},
+		}
+
+		return nil
+	default:
+		return fmt.Errorf("Unknown conversion, got: %T", sink)
+	}
+}
+
+// ConvertFrom implements apis.Convertible.
+// Converts obj from v1beta1.KafkaChannel into v1alpha1.KafkaChannel
+func (sink *KafkaChannel) ConvertFrom(ctx context.Context, obj apis.Convertible) error {
+	switch source := obj.(type) {
+	case *v1beta1.KafkaChannel:
+		statusAddressable := &duckv1alpha1.Addressable{}
+		if source.Status.AddressStatus.Address != nil {
+			if err := statusAddressable.ConvertFrom(ctx, source.Status.AddressStatus.Address); err != nil {
+				return err
+			}
+		}
+
+		subscribableStatus := &eventingduckv1alpha1.SubscribableStatus{
```

Comment:
same here, worth separating in its own ConvertFrom func

Reply:
@nlopezgi you think it is ok to do it later? 
context: https://knative.slack.com/archives/C9JP909F0/p1593764728273200

Reply:
I think this is a proposal, @aliok 

/lgtm
/hold

@nlopezgi if you have no concerns with this step-by-step approach, please cancle the hold